### PR TITLE
Issue 18356 whats changed

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**
This comes to solve https://github.com/dotCMS/core/issues/18356
This code depends on other PR for the enterprise module https://github.com/dotCMS/enterprise-2.x/pull/726

Basically I'm updating the diff tool to use the new HTMLPageAssetRenderedBuilder
The old method doesn't do what it is expected to. Therefore I am deprecating it.

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+        PublishFactory.publishAsset(template, systemUser, false, false);
+    }
+
+    /**
+     * Given scenario: We have a page created out of a layout and a container. The Container holds a List of items.
+     * Expected Result:  We create
+     * @throws Exception
+     */
+    @Test
+    public void Test_Page_With_Changes_Expect_Difference() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:106]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+        when(request.getSession(false)).thenReturn(session);
+        when(request.getSession(true)).thenReturn(session);
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER_ID))
+                .thenReturn(systemUser.getUserId());
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER)).thenReturn(systemUser);
+        when(session.getAttribute(com.dotmarketing.util.WebKeys.CMS_USER)).thenReturn(systemUser);
+
+        when(request.getParameter("host_id")).thenReturn(site.getIdentifier());
+        when(request.getAttribute(WebKeys.CURRENT_HOST)).thenReturn(site);
+        when(session.getAttribute(WebKeys.HTMLPAGE_LANGUAGE)).thenReturn(defaultLang.getId()+"");
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:248]: Codacy Issue found: [Do not add empty strings](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+        final HttpSession session = mock(HttpSession.class);
+
+        when(request.getRequestURI()).thenReturn("/"+pageName);
+        when(request.getSession()).thenReturn(session);
+        when(request.getSession(false)).thenReturn(session);
+        when(request.getSession(true)).thenReturn(session);
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER_ID))
+                .thenReturn(systemUser.getUserId());
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER)).thenReturn(systemUser);
+        when(session.getAttribute(com.dotmarketing.util.WebKeys.CMS_USER)).thenReturn(systemUser);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:244]: Codacy Issue found: [Unnecessary use of fully qualified name 'com.dotmarketing.util.WebKeys.CMS_USER' due to existing import 'com.dotmarketing.util.WebKeys'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+                container.getIdentifier(), contentlet.getIdentifier(),
+                getDotParserContainerUUID(uuid), 0);
+        multiTreeAPI.saveMultiTree(multiTreeV1);
+        HTMLPageDataGen.publish(pageLive);
+
+        //Force a working copy identical to the original
+        final Contentlet workingPage = contentletAPI.checkout(pageLive.getInode(), systemUser, false);
+        contentletAPI.checkin(workingPage, systemUser, false);
+
+        HttpServletRequest request = mock(HttpServletRequest.class);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:233]: Codacy Issue found: [Local variable 'request' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+    private static Folder folder;
+    private static User systemUser;
+    private static Container container;
+    private static Language defaultLang;
+    private static String uuid;
+
+    private static final String NOTHING_CHANGED = "Nothing Changed";
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:69]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+                getDotParserContainerUUID(uuid), 0);
+        multiTreeAPI.saveMultiTree(multiTreeV1);
+        HTMLPageDataGen.publish(pageLive);
+
+        //Force a working copy identical to the original
+        final Contentlet workingPage = contentletAPI.checkout(pageLive.getInode(), systemUser, false);
+        contentletAPI.checkin(workingPage, systemUser, false);
+
+        HttpServletRequest request = mock(HttpServletRequest.class);
+        HttpServletResponse response = mock(HttpServletResponse.class);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:234]: Codacy Issue found: [Local variable 'response' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+        }
+    }
+
+
+    /**
+     *
+     * @throws Exception
+     */
+    @Test
+    public void Test_Page_No_Changes_Expect_Nothing_Changed_Message() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:201]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java [hunk 8]
```diff
@@ -0,0 +1,272 @@
...
+                .nextPersisted();
+
+        folder = new FolderDataGen().site(site).nextPersisted();
+
+        PublishFactory.publishAsset(template, systemUser, false, false);
+    }
+
+    /**
+     * Given scenario: We have a page created out of a layout and a container. The Container holds a List of items.
+     * Expected Result:  We create
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:102]: Expected Result looks incomplete

File: dotCMS/src/main/webapp/html/portlet/ext/htmlpages/view_live_working_diff.jsp [hunk 9]
```diff
@@ -8,17 +8,17 @@
     String id = request.getParameter("id");
     long lang = Long.parseLong(request.getParameter("pageLang"));
     User user = APILocator.getUserAPI().getSystemUser();
-    String contentId = request.getParameter("contentId");
-    IHTMLPage p = null;
+    //String contentId = request.getParameter("contentId");
```
Comment (dsilvam) [inline dotCMS/src/main/webapp/html/portlet/ext/htmlpages/view_live_working_diff.jsp:11]: remove comment

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/webapp/html/portlet/ext/htmlpages/view_live_working_diff.jsp (hunk 9 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/webapp/html/portlet/ext/htmlpages/view_live_working_diff.jsp]: done

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java [hunk 10]
```diff
@@ -0,0 +1,273 @@
...
+                container.getIdentifier(), contentlet.getIdentifier(),
+                getDotParserContainerUUID(uuid), 0);
+        multiTreeAPI.saveMultiTree(multiTreeV1);
+        HTMLPageDataGen.publish(pageLive);
+
+        //Force a working copy identical to the original
+        final Contentlet workingPage = contentletAPI.checkout(pageLive.getInode(), systemUser, false);
+        contentletAPI.checkin(workingPage, systemUser, false);
+
+        HttpServletRequest request = mock(HttpServletRequest.class);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:234]: Codacy Issue found: [Local variable 'request' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java [hunk 11]
```diff
@@ -0,0 +1,273 @@
...
+    }
+
+
+    /**
+     *Given scenario: We have a page created out of a layout and a container. The Container holds a List of items.
+     * Expected Result:  We create a working copy od the page but this one isn't modified. So whn compared
+     * @throws Exception
+     */
+    @Test
+    public void Test_Page_No_Changes_Expect_Nothing_Changed_Message() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:202]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java [hunk 12]
```diff
@@ -0,0 +1,273 @@
...
+        when(request.getSession(false)).thenReturn(session);
+        when(request.getSession(true)).thenReturn(session);
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER_ID))
+                .thenReturn(systemUser.getUserId());
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER)).thenReturn(systemUser);
+        when(session.getAttribute(com.dotmarketing.util.WebKeys.CMS_USER)).thenReturn(systemUser);
+
+        when(request.getParameter("host_id")).thenReturn(site.getIdentifier());
+        when(request.getAttribute(WebKeys.CURRENT_HOST)).thenReturn(site);
+        when(session.getAttribute(WebKeys.HTMLPAGE_LANGUAGE)).thenReturn(defaultLang.getId()+"");
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:249]: Codacy Issue found: [Do not add empty strings](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java [hunk 13]
```diff
@@ -0,0 +1,273 @@
...
+        PublishFactory.publishAsset(template, systemUser, false, false);
+    }
+
+    /**
+     * Given scenario: We have a page created out of a layout and a container. The Container holds a List of items.
+     * Expected Result:  We create a working copy and modify the list of items. The new items must replace the old ones.
+     * @throws Exception
+     */
+    @Test
+    public void Test_Page_With_Changes_Expect_Difference() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:106]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java [hunk 14]
```diff
@@ -0,0 +1,273 @@
...
+        final HttpSession session = mock(HttpSession.class);
+
+        when(request.getRequestURI()).thenReturn("/"+pageName);
+        when(request.getSession()).thenReturn(session);
+        when(request.getSession(false)).thenReturn(session);
+        when(request.getSession(true)).thenReturn(session);
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER_ID))
+                .thenReturn(systemUser.getUserId());
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER)).thenReturn(systemUser);
+        when(session.getAttribute(com.dotmarketing.util.WebKeys.CMS_USER)).thenReturn(systemUser);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:245]: Codacy Issue found: [Unnecessary use of fully qualified name 'com.dotmarketing.util.WebKeys.CMS_USER' due to existing import 'com.dotmarketing.util.WebKeys'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java [hunk 15]
```diff
@@ -0,0 +1,273 @@
...
+                getDotParserContainerUUID(uuid), 0);
+        multiTreeAPI.saveMultiTree(multiTreeV1);
+        HTMLPageDataGen.publish(pageLive);
+
+        //Force a working copy identical to the original
+        final Contentlet workingPage = contentletAPI.checkout(pageLive.getInode(), systemUser, false);
+        contentletAPI.checkin(workingPage, systemUser, false);
+
+        HttpServletRequest request = mock(HttpServletRequest.class);
+        HttpServletResponse response = mock(HttpServletResponse.class);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:235]: Codacy Issue found: [Local variable 'response' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java [hunk 16]
```diff
@@ -0,0 +1,273 @@
...
+    private static Folder folder;
+    private static User systemUser;
+    private static Container container;
+    private static Language defaultLang;
+    private static String uuid;
+
+    private static final String NOTHING_CHANGED = "Nothing Changed";
+
+    @BeforeClass
+    public static void prepare() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:69]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java (hunk 13 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:106]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java (hunk 14 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:245]: Codacy Issue found: [Unnecessary use of fully qualified name 'com.dotmarketing.util.WebKeys.CMS_USER' due to existing import 'com.dotmarketing.util.WebKeys'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java (hunk 10 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:234]: Codacy Issue found: [Local variable 'request' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java (hunk 16 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:69]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java (hunk 12 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:249]: Codacy Issue found: [Do not add empty strings](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java (hunk 11 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:202]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java (hunk 15 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:235]: Codacy Issue found: [Local variable 'response' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java [hunk 17]
```diff
@@ -0,0 +1,273 @@
...
+        when(request.getSession(false)).thenReturn(session);
+        when(request.getSession(true)).thenReturn(session);
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER_ID))
+                .thenReturn(systemUser.getUserId());
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER)).thenReturn(systemUser);
+        when(session.getAttribute(com.dotmarketing.util.WebKeys.CMS_USER)).thenReturn(systemUser);
+
+        when(request.getParameter("host_id")).thenReturn(site.getIdentifier());
+        when(request.getAttribute(WebKeys.CURRENT_HOST)).thenReturn(site);
+        when(session.getAttribute(WebKeys.HTMLPAGE_LANGUAGE)).thenReturn(defaultLang.getId()+"");
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:249]: Codacy Issue found: [Do not add empty strings](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java (hunk 13 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:106]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java [hunk 18]
```diff
@@ -0,0 +1,273 @@
...
+    }
+
+
+    /**
+     *Given scenario: We have a page created out of a layout and a container. The Container holds a List of items.
+     * Expected Result:  We create a working copy od the page but this one isn't modified. So whn compared
+     * @throws Exception
+     */
+    @Test
+    public void Test_Page_No_Changes_Expect_Nothing_Changed_Message() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:202]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java (hunk 16 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:69]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java [hunk 19]
```diff
@@ -0,0 +1,273 @@
...
+        final HttpSession session = mock(HttpSession.class);
+
+        when(request.getRequestURI()).thenReturn("/"+pageName);
+        when(request.getSession()).thenReturn(session);
+        when(request.getSession(false)).thenReturn(session);
+        when(request.getSession(true)).thenReturn(session);
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER_ID))
+                .thenReturn(systemUser.getUserId());
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER)).thenReturn(systemUser);
+        when(session.getAttribute(com.dotmarketing.util.WebKeys.CMS_USER)).thenReturn(systemUser);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:245]: Codacy Issue found: [Unnecessary use of fully qualified name 'com.dotmarketing.util.WebKeys.CMS_USER' due to existing import 'com.dotmarketing.util.WebKeys'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

**Review by nollymar — APPROVED** [review]

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java [hunk 20]
```diff
@@ -0,0 +1,273 @@
...
+    }
+
+
+    /**
+     *Given scenario: We have a page created out of a layout and a container. The Container holds a List of items.
+     * Expected Result:  We create a working copy od the page but this one isn't modified. So whn compared
+     * @throws Exception
+     */
+    @Test
+    public void Test_Page_No_Changes_Expect_Nothing_Changed_Message() throws Exception {
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:202]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java [hunk 21]
```diff
@@ -0,0 +1,273 @@
...
+        when(request.getSession(false)).thenReturn(session);
+        when(request.getSession(true)).thenReturn(session);
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER_ID))
+                .thenReturn(systemUser.getUserId());
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER)).thenReturn(systemUser);
+        when(session.getAttribute(com.dotmarketing.util.WebKeys.CMS_USER)).thenReturn(systemUser);
+
+        when(request.getParameter("host_id")).thenReturn(site.getIdentifier());
+        when(request.getAttribute(WebKeys.CURRENT_HOST)).thenReturn(site);
+        when(session.getAttribute(WebKeys.HTMLPAGE_LANGUAGE)).thenReturn(defaultLang.getId()+"");
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:249]: Codacy Issue found: [Do not add empty strings](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java (hunk 13 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:106]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java (hunk 16 above)
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:69]: Codacy Issue found: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

File: dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java [hunk 22]
```diff
@@ -0,0 +1,273 @@
...
+        final HttpSession session = mock(HttpSession.class);
+
+        when(request.getRequestURI()).thenReturn("/"+pageName);
+        when(request.getSession()).thenReturn(session);
+        when(request.getSession(false)).thenReturn(session);
+        when(request.getSession(true)).thenReturn(session);
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER_ID))
+                .thenReturn(systemUser.getUserId());
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER)).thenReturn(systemUser);
+        when(session.getAttribute(com.dotmarketing.util.WebKeys.CMS_USER)).thenReturn(systemUser);
```
Comment (dev-dotcms) [inline dotCMS/src/integration-test/java/com/dotcms/enterprise/HTMLDiffUtilTest.java:245]: Codacy Issue found: [Unnecessary use of fully qualified name 'com.dotmarketing.util.WebKeys.CMS_USER' due to existing import 'com.dotmarketing.util.WebKeys'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5511524)

**Review by erickgonzalez — APPROVED** [review]

**Pull request merged**